### DISCUSSION TITLE
"Fixing" tracking branches

### DISCUSSION THREAD

**Discussion by BatmanAoD** [jj-vcs/jj#2552] [Q&A] [answered]
I had a tracking branch that I accidentally let diverge from the remote version, so I tried to fix it as follows:

```
jj abandon <branch>
jj git fetch
jj checkout <branch>
```

At this point, I expected `jj diff --from <branch>@origin>` to be empty, but it wasn't. Why not? Should I have used `branch forget` instead of `abandon`?

I ended up just manually calling `jj branch set <branch> -r <branch>@origin`, which is probably what I should have done in the first place.

**Comment by martinvonz** [comment, marked as answer]
> At this point, I expected `jj diff --from <branch>@origin>` to be empty, but it wasn't. Why not? Should I have used `branch forget` instead of `abandon`?

Yes. `abandon` acts on commits, so `jj abandon <branch>` will abandon the commit the branch points to.

> I ended up just manually calling `jj branch set <branch> -r <branch>@origin`, which is probably what I should have done in the first place.

Yes, that makes sense.
